scripts/check-pricing

```diff
@@ -0,0 +1,133 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+print_help() {
+  cat <<-EOF
+	Checks that API pricing covers cycle costs.
+
+	For each API method, this:
+	* Gets the signer canister cycles balance
+	* Makes a sample API call
+	* Gets the cycles balance again
+	* Reports on the difference.
+
+	Note: Measurement is affected by other API calls made to the network.
+
+	Usage: "$(basename "$0")" [network]
+	EOF
+}
+
+[[ "${1:-}" != "--help" ]] || {
+  print_help
+  exit 0
+}
+
+DFX_NETWORK="${1:-ic}"
+SIGNER_CANISTER_ID="$(dfx canister id --network "$DFX_NETWORK" signer)"
+
+to_number() {
+  : Converts symbols such as 1K to 1_000
+  numfmt --from=si | rev | sed 's/.../&_/g' | rev
+}
+signer_balance() {
+  dfx canister status --network "$DFX_NETWORK" signer | awk '($1 == "Balance:"){print $2}'
+}
+check_call_pricing() {
+  local before after
+  before="$(signer_balance)"
+  dfx canister call --network "$DFX_NETWORK" signer "${@}" >/dev/null
+  after="$(signer_balance)"
+  diff="$(echo "$after - $before" | tr -d _ | bc)"
+  if [[ $diff = -* ]]; then
+    echo "WARNING: signer balance fell by $diff for: $1"
+  else
+    echo "OK: Signer balance rose by $diff for: $1"
+  fi
+}
+
+APPROVAL="1P"
+
+: Approve funds for use in the test
+dfx canister call cycles_ledger icrc2_approve --network "$DFX_NETWORK" "
+  record {
+    amount = $(echo "$APPROVAL" | to_number);
+    spender = record {
+      owner = principal \"${SIGNER_CANISTER_ID}\";
+    };
+  }" >/dev/null
+
+: Check that each method is making a profit
+
+check_call_pricing btc_caller_address '
+(
+  record { network = variant { mainnet }; address_type = variant { P2WPKH } },
+  opt variant { CallerPaysIcrc2Cycles },
+)
+'
+check_call_pricing btc_caller_balance '
+
+(
+  record { network = variant { mainnet }; address_type = variant { P2WPKH } },
+  opt variant { CallerPaysIcrc2Cycles },
+)
+'
+check_call_pricing btc_caller_send '
+(
+  record {
+    fee_satoshis = null;
+    network = variant { mainnet };
+    utxos_to_spend = vec {};
```

Comment:
This is the scenario where there are less cycles consumed, no? With no utxos to spend, it won't iterate over the, nor sign any output without any output.

Do you plan on running this against mainnet?